packages/astro/src/actions/runtime/virtual/shared.ts

```diff
@@ -81,8 +81,8 @@ export class ActionError<T extends ErrorInferenceObject = ErrorInferenceObject>
 			return new ActionError(body);
 		}
 		return new ActionError({
-			message: res.statusText,
-			code: ActionError.statusToCode(res.status),
+			message: body.message,
+			code: ActionError.statusToCode(body.status),
```

Comment:
Should we also anticipate if `body` is not an object? Or if the property access returns undefined?

Reply:
Yeah that's a good call. We should add a check

Reply:
Update: I just realized this was the base case when an Action error code not be parsed. I decided to simply return `INTERNAL_SERVER_ERROR` for that case, since that is unexpected.